.github/workflows/docs.yaml

```diff
@@ -0,0 +1,67 @@
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "!**-rc"
+    paths:
+      - docs/**
+      - .github/workflows/docs.yml
+
+
+name: Deploy Datafusion Python site
+
+jobs:
+  build-docs:
+    name: Build docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set target branch
+        id: target-branch
+        run: |
+          if test '${{ github.ref }}' = 'refs/heads/master'; then
+            echo "value=asf-staging" >> $GITHUB_OUTPUT
+          elif test '${{ github.ref_type }}' = 'tag'; then
+            echo "value=asf-site" >> $GITHUB_OUTPUT
+          fi
+      - name: Checkout docs sources
+        uses: actions/checkout@v3
+      - name: Checkout docs target branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ steps.target-branch.outputs.value }}
+          path: docs-target
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install dependencies
+        run: |
+          python3 -m venv venv
+          source venv/bin/activate
+          pip install -r requirements-310.txt
+          pip install -r docs/requirements.txt
+      
+      - name: Build Datafusion
+        run: |
+          source venv/bin/activate
+          maturin develop
+
+      - name: Build docs
+        run: |
+          source venv/bin/activate
+          cd docs
+          make html
+
+      - name: Copy & push the generated HTML
+        run: |
+          cd docs-target
+          find ./ -name "*.html" -type f -print0 | xargs -0 /bin/rm -f
```

Comment:
This may not work for other extensions such as `*.css`. How about using `rsync`?

```shell
rsync \
  -a \
  --delete \
  --exclude '/.git/' \
  ../docs/build/html/ \
  ./
```

Reply:
Right! There are .css/.js/images/... !
I have to test your approach!
The important thing is to preserve anything that is not autogenerated, especially `.asf.yaml`

Reply:
How about copying `.asf.yaml` from `master` to `asf-site`?
apache/arrow-site uses this approach: https://github.com/apache/arrow-site/pull/116

Reply:
This will work for asf-site, but it will break asf-staging, because it uses `staging:` instead of `publish:`

Reply:
> This will work for asf-site, but it will break asf-staging, because it uses `staging:` instead of `publish:`

Can we write both `staging:` and `publish:` in `master`'s `.asf.yaml`?

```yaml
staging:
  whoami: asf-staging
  subdir: datafusion-python
  
publish:
  whoami: asf-site
  subdir: datafusion-python
```

Reply:
I think this should work!
The only way to find out is to try it! :-)

Reply:
I think so too. :-)

Reply:
done with d5e32a2